packages/lesswrong/server/vulcan-lib/apollo-ssr/pageCache.ts

```diff
@@ -167,7 +167,7 @@ const cacheLookup = (cacheKey: string, abTestGroups: CompleteTestGroupAllocation
 
 const objIsSubset = <A extends Record<string, any>, B extends Record<string, any>>(subset: A, superset: B): boolean => {
   for (let key in subset) {
-    if (!(key in superset) || subset[key] !== superset[key])
+    if (!(key in superset) || subset[key] !== superset[key as AnyBecauseHard])
```

Comment:
I continue to be annoyed that `in` (checking whether a property exists in an object) only narrows types for string literals, but I'm confused by why this used to work but no longer does.  Does `for (let key in subset)` narrow the type of `key` to `keyof A` where it didn't before?